tests/emukit/core/test_outer_loop.py

```diff
@@ -68,48 +76,87 @@ def X(self):
         def Y(self):
             return self._Y
 
-        def predict(self, X):
+        def predict(self, x):
             pass
 
-        def update_data(self, X, Y):
-            self._X = X
-            self._Y = Y
+        def update_data(self, x, y):
+            self._X = x
+            self._Y = y
 
         def optimize(self):
             pass
 
-    model = MockModel()
-    model_updater = MockModelUpdater(model)
+    mock_model = MockModel()
+    model_updater = MockModelUpdater(mock_model)
 
     loop = OuterLoop(mock_next_point_calculator, model_updater)
     loop.run_loop(mock_user_function, 2)
 
     # Check update was last called with a loop state with all the collected data points
-    assert model.X.shape[0] == 2
-    assert model.Y.shape[0] == 2
+    assert mock_model.X.shape[0] == 2
+    assert mock_model.Y.shape[0] == 2
+
 
 def test_accept_non_wrapped_function(mock_next_point_calculator, mock_updater):
     stopping_condition = mock.create_autospec(StoppingCondition)
     stopping_condition.should_stop.side_effect = [False, False, True]
-    user_function = lambda x: np.array([[0]])
+
+    def user_function(x):
+        return np.array([[0]])
 
     loop = OuterLoop(mock_next_point_calculator, mock_updater)
     loop.run_loop(user_function, stopping_condition)
 
-    assert(loop.loop_state.iteration == 2)
-    assert(np.array_equal(loop.loop_state.X, np.array([[0], [0]])))
+    assert (loop.loop_state.iteration == 2)
+    assert (np.array_equal(loop.loop_state.X, np.array([[0], [0]])))
+
 
 def test_default_condition(mock_next_point_calculator, mock_updater, mock_user_function):
     n_iter = 10
 
     loop = OuterLoop(mock_next_point_calculator, mock_updater)
     loop.run_loop(mock_user_function, n_iter)
 
-    assert(loop.loop_state.iteration == n_iter)
+    assert (loop.loop_state.iteration == n_iter)
+
 
 def test_condition_invalid_type(mock_next_point_calculator, mock_updater, mock_user_function):
     invalid_n_iter = 10.0
     loop = OuterLoop(mock_next_point_calculator, mock_updater)
 
     with pytest.raises(ValueError):
         loop.run_loop(mock_user_function, invalid_n_iter)
+
+
+def test_custom_step_call():
```

Comment:
this is no longer called "custom_step". how about "test_iteration_end_event"?

Reply:
Renamed